src/mesh/voxel_cloud.rs

```diff
@@ -448,6 +450,250 @@ impl ScalarField {
+                            .iter()
+                            .map(|a| {
+                                self.value_at_absolute_voxel_coordinate(&a)
+                                    .map(|value| volume_value_range.contains(&value))
+                                    .unwrap_or(false)
+                            })
+                            .collect();
+
+                    // The marching cubes lookup table contains 256 possible
+                    // samples of mesh patches. The sample index specifies the
+                    // proper sample for the current voxel group.
+                    let mut sample_index = 0_u8;
+                    if corners_inside_volume_pattern[0] {
+                        sample_index |= 1;
+                    }
+                    if corners_inside_volume_pattern[1] {
+                        sample_index |= 2;
+                    }
+                    if corners_inside_volume_pattern[2] {
+                        sample_index |= 4;
+                    }
+                    if corners_inside_volume_pattern[3] {
+                        sample_index |= 8;
+                    }
+                    if corners_inside_volume_pattern[4] {
+                        sample_index |= 16;
+                    }
+                    if corners_inside_volume_pattern[5] {
+                        sample_index |= 32;
+                    }
+                    if corners_inside_volume_pattern[6] {
+                        sample_index |= 64;
+                    }
+                    if corners_inside_volume_pattern[7] {
+                        sample_index |= 128;
+                    }
+
+                    // The marching cubes lookup table sample, that contains
+                    // triplets of indices to the edge midpoints (currently
+                    // already vertices) that will become mesh patch faces.
+                    let marching_cubes_faces =
+                        MARCHING_CUBES_LOOKUP_TABLE[usize::from(sample_index)].as_slice();
+
+                    // In some cases (when all the voxels are inside or outside
+                    // a volume), there is no mesh patch to be created.
+                    if !marching_cubes_faces.is_empty() {
+                        marching_cubes.push(
+                        // Generate new mesh path from all edge midpoint
+                        // vertices and only those faces that should be created
+                        // according to the marching cubes lookup table.
+                        //
+                        // FIXME: @Optimization remove unused vertices already
+                        // here and generate the mesh without the need of
+                        // removing orphans.
+                        Mesh::from_triangle_faces_with_vertices_and_computed_normals_remove_orphans(
+                            marching_cubes_faces.iter().map(|(a, b, c)| (u32::from(*a), u32::from(*b), u32::from(*c))),
```

Comment:
If you format this manually, does rustfmt return it to this style? It has a tendency to break for large files or large level of nesting.